File: frameworks/PHP/hypervel/app/Providers/RouteServiceProvider.php
<?php

declare(strict_types=1);

namespace App\Providers;

use Hypervel\Foundation\Support\Providers\RouteServiceProvider as BaseServiceProvider;
use Hypervel\Support\Facades\Route;

class RouteServiceProvider extends BaseServiceProvider
{
    /**
     * The route files for the application.
     */
    protected array $routes = [
    ];

    public function boot(): void
    {
        parent::boot();

        Route::group(
            '/api',
            base_path('routes/api.php'),
            ['middleware' => 'api']
        );

        Route::group(
            '/',
            base_path('routes/web.php'),
            ['middleware' => 'web']
        );
    }
}

File: frameworks/Go/goravel/src/gin/app/providers/route_service_provider.go
package providers

import (
	"github.com/goravel/framework/contracts/foundation"

	"goravel/routes"
)

type RouteServiceProvider struct{}

func (receiver *RouteServiceProvider) Register(app foundation.Application) {
}

func (receiver *RouteServiceProvider) Boot(app foundation.Application) {
	routes.Web()
}
